### DISCUSSION TITLE
getEdgeProps doesn't return content from params

### DISCUSSION THREAD

**Discussion by augustomallmann** [flareact/flareact#229] [Q&A] [answered]
Hi, it's me again!

I'm trying to get the `lang` param from the following route:
`./pagens/[lang]/blog/index.tsx`

But, when I try to fetch content based on the param `lang`, the function always throws errors. So I decided to debug and found out that it was returning an empty object. For example:

```
export async function getEdgeProps({ params }) {
    const { lang } = params;
    const content = await getCategoryContent(lang, formatSlug);

    return {
        props: {
            params,
            content,
        }
    };
}
```

When `console.log` the `params`, it returns `{}`. 

Is there a way to fetch content based on route params in a regular route? Taking advance of this topic, do you know any i18n lib that works with flareact rendering on the edge?

**Comment by tj-kev** [comment, marked as answer]
Hi,
looks like this is a missing feature which really should be implemented.

In the meantime, you could get it from the event pathname and extract it from there.

```
export async function getEdgeProps({ event }) {
    const { pathname } = new URL(event.request.url);

   ...
}
```

**Reply by traffisco** [reply]
What do you mean not implemented? Which part?
it should work. I think...
https://github.com/flareact/flareact/blob/canary/tests/pages.spec.js#L100

**Reply by tj-kev** [reply]
Sorry, you're right. I just did a quick test at the time and had the same results as @augustomallmann.

Looking into it a bit further though it looks like its dependent on there being a [slug].js file rather than index.js.

The following works (with [slug].js):
```
/test/[myparam]/anotherdir/[slug]
```

Result:
```
{
  myparam: 'my-param-here',
  slug: 'my-slug-here'
}
```

The following doesn't work (with index.js):
```
/test/[myparam]/anotherdir/
```

Result:
```
{}
```

**Reply by traffisco** [reply]
You right, apparently its really a missing feature which really should be implemented. It shouldn't be depended on [slug].js
